### ISSUE TITLE
'keyCode' does not work with (keyIsPressed) event & 'key' does not work with function keyPressed() 

### ISSUE THREAD

**Issue by mrbombmusic** [processing/p5.js#2110]
#### Nature of issue?

- [x ] Found a bug

#### Most appropriate sub-area of p5.js?

- [x ] Events

#### Which platform were you using when you encountered this?

- [ x] Desktop/Laptop

#### Details about the bug: 

- p5.js version: 0.6.2
- Web browser and it's version: Chrome - Version 60.0.3112.90 (Official Build) (64-bit) (Have encounter this issue on web editor as well as downloaded version)
- Operating System: OSX El Capitan v. 10.11.6

I have encountered  that when using key press events with (keyIsPressed) inside draw(), the code will not execute using the keyCode == property  whereas is will work with key == property.

***example***

function setup() {
  createCanvas(400, 400);
  background(255);
}

function draw() {

  if (keyIsPressed) {
  //  if (key == 'a') {     // ellipse code does execute when 'a' key is pressed
  //  if(keyCode == 65) {   // ellipse code does not execute when 'a' key is pressed
      ellipse(200, 200, 100, 100);
    }
  }
}

Inversely, when using key press events with function keyPressed(), the code will execute using the keyCode == property  whereas it will not work with key == property.

***example***

function setup() {
  createCanvas(400, 400);
  background(255);
}

function draw() {

}

function keyPressed() {
  //   if (keyCode == 65) {       // ellipse code does execute when 'a' key is pressed
 //     if (key == 'a') {         // ellipse code does not execute when 'a' key is pressed
    ellipse(200, 200, 100, 100);
  }
}

**Comment by ajrussellaudio** [comment]
This is all expected behaviour.

`keyIsPressed` isn't an event, it's just a boolean. Meanwhile `key` always refers to the _last_ key pressed, even when no key is currently pressed. So in your first example, `key == 'a'` would evaluate to `true` even if the outer `if` condition was `false`.

The `keyPressed` event is a little more subtle. When you type a key, three events are fired in this order: `keyPressed`, `keyTyped`, and `keyReleased`.

* `keyPressed` fires as soon as the physical key button is down (more or less). It doesn't really know what key is pressed, unless the key is a 'special' key like Shift or an arrow key, which p5 gives you access to in the `keyCode` variable. For keys containing a character, the OS lets the actual keyboard itself figure out what key is pressed and report back a little later.
* `keyTyped` is fired when they keyboard reports _which_ key has been typed. This is where you would have access to the key's character value, and is where the global `key` variable is set.
* `keyReleased` is also aware of the key's value and the `key` variable, but only fires when the key button returns up. 

If you hold a key down, `keyPressed` and `keyTyped` can fire multiple times, but `keyReleased` will only fire once.

**tl;dr** You want the `keyTyped` event in your second example and not `keyPressed`

**Comment by mrbombmusic** [comment]
Thank you very much for the explanation. I do a lot of work with interactive sketches using key presses, so this information is very helpful.

**Comment by RAHULTATA10** [comment]
ok
